Replace waitAndAssert in config_test.go with poll.WaitOn

yongtang opened the pull request:
This fix replaces waitAndAssert in config_test.go with poll.WaitOn
so that the testing is consistent with all other tests in integration.

Also, config_test.go uses to wait and sleep for 2 * (1 minutes) to get the task
info. This fix combined those two sleep and wait for 1 mins. Think 1 min
is enough for config test.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/moby/moby/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


**- A picture of a cute animal (not mandatory but encouraged)**

thaJeztah commented:
oh! in case you were thinking on moving the `SwarmLockUnlock` tests to `integration/`. I had this written for the NodeStatus; https://github.com/moby/moby/compare/master...thaJeztah:poll_node 😅

olljanat commented:
Derek add label: rebuild/*

thaJeztah commented on integration/config/config_test.go:
```diff
@@ -267,19 +268,32 @@ func TestTemplatedConfig(t *testing.T) {
-		return task.NodeID != "" && task.Status.ContainerStatus != nil && task.Status.ContainerStatus.ContainerID != ""
-	})
+	}
+	poll.WaitOn(t, getServiceTask, swarm.ServicePoll, poll.WithTimeout(1*time.Minute))
```
I wonder if we should split this into two steps;

- check if the service reconciled (check if the desired number of tasks is running); that may be more generic / reusable
- once that's the case, do the `inspect` on its task

thaJeztah reviewed (changes requested):
Left some ideas; actually part of that code written while reviewing earlier today, so I pushed it as a branch; https://github.com/moby/moby/pull/38558 😅

yongtang committed:
Replace waitAndAssert in config_test.go with poll.WaitOn

This fix replaces waitAndAssert in config_test.go with poll.WaitOn
so that the testing is consistent with all other tests in integration.

Also, config_test.go uses to wait and sleep for 2 * (1 minutes) to get the task
info. This fix combined those two sleep and wait for 1 mins. Think 1 min
is enough for config test.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang commented:
Thanks @thaJeztah for the review. I have updated the PR and used the changes you made to `swarm.GetRunningTasks`. Please take a look

thaJeztah reviewed (approved):
LGTM, thanks!

vdemeester reviewed (approved):
LGTM :tiger:
